content/fundamentals/lifecycle-events.md

```diff
@@ -81,6 +83,8 @@ async function bootstrap() {
 bootstrap();
 ```
 
+> info **Info** `enableShutdownHooks` is disabled by default because the application could run into memory leak issues in case multiple Nest instances are running (e.g. testing). Node.js would warn you, in case you have too many listeners running in parallel and prevent you from having major memory leaks.
```

Comment:
@BrunnerLivio Could you clarify a couple of things please?
1. Is the memory leak issue **only** a concern in testing?  I.e., there may be many scenarios besides testing where multiple Nest instances are running, and the way this is written might leave people concerned about memory leaks.  If it's not "just in a testing scenario" that this happens, are we able to provide any additional guidelines? 
2. I don't quite understand this sentence: "Node.js would warn you, in case you have too many listeners running in parallel and prevent you from having major memory leaks.".  Does this warning happen automatically if Node detects too many listeners (i.e., the testing scenario above)?  By saying "Node.js **would warn you**", it suggests that this warning would happen "under some certain conditions", though it's not clear to me when that warning would happen  And are we saying "hey, don't worry, you're OK as long as you don't get this warning"?


Sorry to nitpick!

Reply:
> Is the memory leak issue only a concern in testing? 
> "Node.js would warn you, in case you have too many listeners running in parallel and prevent you from having major memory leaks.". 


No - so basically, what `enableShutdownHooks` just starts listeners on all shutdown hooks. As far as I remember, it's around seven listeners. In case you would run e2e tests with `enableShutdownHooks()` on and you would have parallel tests enabled (which I think is the case by with Jest default), then Node.js would complain that you have too many listeners in one Node process.
Of course, this could also occur in non-test-environment circumstances, in case you fire up multiple Nest apps with `enableShutdownHooks` in one process.
 
So I guess its technically not a "memory leak". How shall we phrase this?

Reply:
Ah, OK.  What if we have a somewhat general statement warning.  I'm not sure I know quite enough to word this, but along the lines of:

"`enableShutdownHooks` consumes memory by starting listeners.  In cases where you are running multiple Nest apps in a single Node process (e.g., when running parallel tests with Jest), Node may complain about excessive listener processes.  For this reason, `enableShutdownHooks` is not enabled by default.  Be aware of this condition when you are running multiple instances in a single Node process."

Not perfect, but maybe if you agree with that you can propose something like it?

Reply:
LGTM. Added it in!